### PR TITLE
Javascript Capstone Project- tv webapp

### PR DESCRIPTION
**Pull request by LeslieAine**
In this project, we created a tv webapp showing a list of movies that allows users to comment and like different movies. We took the following steps in creating this project;

-We created the main part of the homepage that keeps the layout from the wireframe and when the page loads, the tv webapp retrieves data from the https://www.tvmaze.com/api API and shows the list of items on the screen with the basic data title and image.
-We prepared the "Comments" and "Likes" buttons.
- When the user clicks on the Like button of an item (on Homepage), the interaction is recorded in the Involvement API and the screen is updated.
- When the page loads, the webapp shows the item likes and combines them with the data from the base API.

- We added the movies counter to show number of movies displayed

-We created a comments popup only with the top section (displaying details of the selected movie) such that when the popup loads, the web app retrieves data from the Involvement API to show the movie comments.

- We implemented the "add new comment" form on the Comments pop-up according to the required style and created a specific function to calculate the number of comments the API gave us. 

- We made sure that the correct number is displayed on the Comments pop up.
- We made sure that the function for the comments counter is covered with unit tests using Jest.
-We made sure that we have tests for all edge cases.

Below is a video describing the project steps;
https://drive.google.com/file/d/1K09FVIRjMmHYnRSKgV42xpTgl7nuKPYe/view?usp=sharing

### CODE REVIEW

File: src/modules/involvementAPI/addLikes.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  likesContainer.forEach((like) => {
+    const id = like.parentElement.parentElement.parentElement.parentElement
+      .firstElementChild.textContent;
+    const movieLikes = likes.filter(
+      (item) => parseInt(item.item_id, 10) === parseInt(id, 10),
+    );
+    like.textContent = movieLikes.length === 0 ? 0 : movieLikes[0].likes;
+  });
+};
+
```
Comment (ghost) [inline src/modules/involvementAPI/addLikes.js:15]: - [ ] I think it would be better that if a user clicks on the `like` button, the  `likes counter` increments instantly without having to refresh the page to see the change. This can be achieved by fetching data from the `involvement API` whenever the `like` button is clicked. This can help improve the user experience.  Also, you could cover the `likes counter`  in your tests as it is required in this milestone that all counters should be tested
- [ ] The `live demo` link on your readme file is broken. If you have not deployed your project yet, Kindly remove it to keep your repo professional

**Review by ghost — COMMENTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the assessment but you are almost there!





## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/modules/involvementAPI/addLikes.js (hunk 1 above)
Comment (LeslieAine) [reply src/modules/involvementAPI/addLikes.js:15]: Thank you Frank for the review. 
We have fixed the page to show likes increment on clicking. However, testing for the likes counter was not included in the project cards for the required activities.

File: README.md [hunk 2]
```diff
@@ -1 +1,169 @@
...
+- **Lets users see the number of likes and comments on each series**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Here](https://LeslieAine.github.io/tv-webapp/)
+
```
Comment (joskalenda) [inline README.md:79]: - [ ] Hey @LeslieAine, I understand the time and energy you investigate in order to deploy the app but this demo link is not working yet, please take your time to fix this as it is very important for you.

File: src/modules/involvementAPI/addLikes.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { getLikes } from './getLikes.js';
+
+const addLikes = async () => {
+  const likes = await getLikes();
+  const likesContainer = document.querySelectorAll('.likes');
+  likesContainer.forEach((like) => {
+    const id = like.parentElement.parentElement.parentElement.parentElement
```
Comment (joskalenda) [inline src/modules/involvementAPI/addLikes.js:7]: - [ ] And yes, what was mentioned by the previous reviewer is true, you need to add tests for all counter functions as per the requirement, I would suggest you implement the change as he mentioned before(test cases) and resubmit your work.

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @LeslieAine , |
|Great work on making the changes requested by a previous reviewer 👏, You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. |
|  |

## Highlights

- [ ]... ✅ Great design
- [ ].. ✅ Green linters
- [ ].... ✅ Great readme file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @LeslieAine @TayyabKM 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
